modules/carto/src/layers/carto-layer.js

```diff
@@ -137,8 +140,12 @@ export default class CartoLayer extends CompositeLayer {
       apiVersion === API_VERSIONS.V3 &&
       format === FORMATS.TILEJSON &&
       new URLSearchParams(data.tiles[0]).get('format');
+    const nonMvtTiles = tilesFormat && tilesFormat !== TILE_FORMATS.MVT;
+
+    const dynamicTiles =
```

Comment:
Not for this PR, but I feel we can make it simpler if maps API always returns the format at tiles url. 

Don't you think?

Reply:
I feel this part is getting complicated

Reply:
if (
      apiVersion === API_VERSIONS.V1 ||
      apiVersion === API_VERSIONS.V2 || 
    )
   {
      // backward compatibility CARTO 2
      layer = MVTLayer;
    } else if (format === FORMATS.TILEJSON) {
       layer = tilesFormat === TILE_FORMATS.MVT ? MVTLayer : CartoDynamicTileLayer
    }
     else {
      layer = GeoJsonLayer;
    }

Reply:
If the backend always going to return the format this suggested code is easier to understand.

Reply:
Yes I would much prefer if the backend returned the format it was sending so we could simplify here